### PR TITLE
Feature/select arbitrator

### PR DESCRIPTION
**Pull request by blabno**

### CODE REVIEW

File: src/test/java/io/bisq/api/ArbitratorResourceIT.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    @InSequence(1)
+    @Test
+    public void registerArbitrator() throws InterruptedException {
+        final int alicePort = getAlicePort();
+        final int arbitratorPort = getArbitratorPort();
+        given().
+                port(alicePort).
+//
```
[the remark this answers is not in the document]
Comment (blabno) [reply src/test/java/io/bisq/api/ArbitratorResourceIT.java]: Bisq code style indents back subsequent line so `then()` is pulled left and it helps distinguish sections of that chain.

File: src/test/java/io/bisq/api/ArbitratorResourceIT.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        when().
+                get("/api/v1/arbitrators").
+//
+        then()
+                .statusCode(200).
+                and().body("arbitrators.size()", equalTo(1)).
+                and().body("total", equalTo(1)).
+                and().body(containsString(":3335"));
```
[the remark this answers is not in the document]
Comment (blabno) [reply src/test/java/io/bisq/api/ArbitratorResourceIT.java]: All I know for sure is the port number. IP might differ depending on docker configuration.
REST arbitrator object contains only "address" property.

File: src/main/java/io/bisq/api/service/v1/ArbitratorResource.java [hunk 3]
```diff
@@ -34,9 +36,33 @@ public void register(ArbitratorRegistration data) {
...
+    }
+
+    @ApiOperation("Deselect arbitrator")
+    @POST
+    @Path("/{address}/deselect")
+    public ArbitratorList deselectArbitrator(@NotBlank @PathParam("address") String address) {
+        return toRestModel(bisqProxy.deselectArbitrator(address));
+    }
+
+    private static ArbitratorList toRestModel(Collection<io.bisq.core.arbitration.Arbitrator> businessModelList) {
```
[the remark this answers is not in the document]
Comment (blabno) [reply src/main/java/io/bisq/api/service/v1/ArbitratorResource.java]: If BisqProxy is something that should be embedded in exchange during refactoring then imho it doesn't want to speak REST domain.

[the remark this answers is not in the document]
Comment (mrosseel) [reply src/main/java/io/bisq/api/service/v1/ArbitratorResource.java]: the idea is that the API models are rest-agnostic, maybe that wasn't clear or has been violated before, but that was the idea.

**Comment by blabno** [comment]
So you want to introduce new layer just for the proxy?

5 mar 2018 16:02 "mrosseel" <notifications@github.com> napisał(a):

File: src/test/java/io/bisq/api/ArbitratorResourceIT.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mrosseel) [reply src/test/java/io/bisq/api/ArbitratorResourceIT.java]: ok

File: src/test/java/io/bisq/api/ArbitratorResourceIT.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrosseel) [reply src/test/java/io/bisq/api/ArbitratorResourceIT.java]: ok

**Comment by mrosseel** [comment]
The model layer should be rest-agnostic, and it is for the most part. In the cases where the Model contains clear REST-isms (e.g. paging fields) that should be a rest-specific model. But let's discuss.

**Comment by blabno** [comment]
Ok, but the core model is rest agnostic, so why can't we use it in BisqProxy and even in Rest?
I think Rest layer should receive core model from BisqProxy (which after exchange refactoring will be dissolved in core business components like services or managers).  
And it should be the Rest layer who is responsible for translation between both models.
If we introduce a different type of consumer layer i.e. ZeroMQ then ZeroMQ should be responsible for domain translation.

**Comment by mrosseel** [comment]
that actually sounds reasonable. Could you also respond to the first comment? thx

File: src/main/java/io/bisq/api/BisqProxy.java [hunk 4]
```diff
@@ -49,6 +49,7 @@
 import org.bitcoinj.core.Transaction;
 
 import javax.annotation.Nullable;
+import javax.ws.rs.BadRequestException;
```
[the remark this answers is not in the document]
Comment (blabno) [reply src/main/java/io/bisq/api/BisqProxy.java]: I agree with you. Refactoring under way.

**Pull request merged**